Core/CookieStorage.swift

```diff
@@ -28,57 +28,13 @@ public class CookieStorage {
 
     private var userDefaults: UserDefaults
 
-    var cookies: [HTTPCookie] {
-
-        var storedCookies = [HTTPCookie]()
-        if let cookies = userDefaults.object(forKey: Constants.key) as? [[String: Any?]] {
-            for cookieData in cookies {
-                var properties = [HTTPCookiePropertyKey: Any]()
-                cookieData.forEach({
-                    properties[HTTPCookiePropertyKey(rawValue: $0.key)] = $0.value
-                })
-
-                if let cookie = HTTPCookie(properties: properties) {
-                    os_log("read cookie %s %s %s", log: generalLog, type: .debug, cookie.domain, cookie.name, cookie.value)
-                    storedCookies.append(cookie)
-                }
-            }
-        }
-
-        return storedCookies
-    }
-
     public init(userDefaults: UserDefaults = UserDefaults.standard) {
         self.userDefaults = userDefaults
     }
 
-    func clear() {
+    public func clear() {
         userDefaults.removeObject(forKey: Constants.key)
         os_log("cleared cookies", log: generalLog, type: .debug)
     }
```

Comment:
So something occurs to me:

1) Migration is unnecessary because we always have a webview, OR 
2) We shouldn't be recreating the webview so quickly after the fire button is used (put a breakpoint on TabViewController.swift:303 and hit the fire button)

Reply:
There is a slight chance cookies list this won't be empty, exactly due to #2 you raised: if cookies were stored with delay and after WebView has been created. It is unlikely, but I still wonder if we should try to preserve those in such case...

Reply:
Fair point @bwaresiak, there's certainly a slim chance where cookies have been stored but not yet consumed. I've restored the `consumeCookies` logic to ensure that any stored cookies are being restored, the same way we've always been doing. Some time in the future, we can remove it again.

Reply:
I just think it's so edge case it's not worth having it in the code base.  I think we'd be better off addressing the issue that there's a webview constructed during the process of deleting data.

Reply:
I think in such case we should value potential consistency over cleaning up codebase - we can still remove this migration as a follow up task scheduled few months from now. 

@brindy WebView construction seem to be a separate issue: as far as I see, this is about when to construct WebView for newly created tabs (and cleaning data's side effect is recreating first tab). I think we opted in to do that immediately, as doing it in a lazy manner can result is noticeable lag on first navigation - still, I'm more than happy to discuss that in Asana if you think we should revisit that and it should happen as a separate project.

Reply:
Yeah, you're right, let's clean it up in a few months.